infrastructure/modules/serverless/cloudtrail.tf

```diff
@@ -14,85 +14,68 @@
 
 # Create an S3 bucket for storing CloudTrail logs
 resource "aws_s3_bucket" "this" {
-  #ts:skip=AWS.S3Bucket.LM.MEDIUM.0078 We are aware of the risk and choose to skip this rule
-  count = var.create_kms ? 0 : 1
+  count = var.create_s3_bucket ? 1 : 0
 
   bucket = "${var.namespace}-cloudtrail-logs-storage"
   tags   = merge(local.tags, tomap({ "Name" = "${var.namespace}-bucket" }))
-
 }
 
+# Create a CloudTrail trail and enable logging to the S3 bucket
+resource "aws_cloudtrail" "this" {
+  count = var.create_cloudtrail ? 1 : 0
 
-resource "aws_s3_bucket_versioning" "bucket_versioning" {
-
-  count = var.create_kms ? 1 : 0
+  depends_on                    = [aws_s3_bucket.this]
+  name                          = "${var.namespace}-cloudtrail"
+  s3_bucket_name                = var.create_s3_bucket ? aws_s3_bucket.this[0].id : null
+  s3_key_prefix                 = "cloudtrail"
+  include_global_service_events = true
+  is_multi_region_trail         = true
+  enable_log_file_validation    = true
+  kms_key_id                    = var.create_kms ? aws_kms_key.this[0].arn : null
 
-  bucket = aws_s3_bucket.this[0].id
-  versioning_configuration {
-    status = "Enabled"
+  event_selector {
```

Comment:
@yaseenzafar123 What does event_selector do?

Reply:
It defines the type of events that we can capture through cloudtrail.

Reply:
Added in 621f2fb80404ba20e621b516b17cd84b11fbf1bb